src/features/call/components/SurveyAccordion.tsx

```diff
@@ -0,0 +1,154 @@
+  ) => {
+    setResponses((prevResponses) => ({
+      ...prevResponses,
+      [questionId]: Array.isArray(value)
+        ? { options: value }
+        : { response: String(value) },
+    }));
+  };
+
+  const handleFormSubmission = () => {
+    const signature = target.id;
+    const formResponses: {
+      [key: number]: { options?: number[]; response?: string };
+    } = responses;
+
+    const mappedResponses: ZetkinSurveyQuestionResponse[] = Object.keys(
+      formResponses
+    )
+      .map((key) => {
+        const questionId = parseInt(key);
+        const response = formResponses[questionId];
+
+        if (response.response) {
+          return {
+            question_id: questionId,
+            response: response.response,
+          };
+        } else if (response.options) {
+          return {
+            options: response.options,
+            question_id: questionId,
+          };
+        }
+
+        return null;
+      })
+      .filter(
+        (response): response is ZetkinSurveyQuestionResponse =>
+          response !== null
+      );
+
+    addSubmission({
+      responses: mappedResponses,
+      signature,
+    });
+  };
+
+  return (
+    <Suspense fallback={<ZUILogoLoadingIndicator />}>
+      <Accordion>
+        <AccordionSummary expandIcon={<ExpandMoreIcon />}>
+          <Typography>{survey.title}</Typography>
+        </AccordionSummary>
+        <AccordionDetails>
+          <Box mb={2}>
+            <Typography variant="h6">{survey.title}</Typography>
+            <Typography>{survey.info_text}</Typography>
+          </Box>
+          {elements &&
+            elements.length > 0 &&
+            elements.map((element) => {
+              if (element.type === ELEMENT_TYPE.QUESTION) {
+                if (element.question.response_type === 'text') {
+                  return (
+                    <SurveyText
+                      key={element.id}
+                      element={element as ZetkinSurveyTextQuestionElement}
+                      onChange={(value) => {
+                        handleResponseChange(element.id, value);
+                      }}
+                    />
+                  );
+                } else if (element.question.response_type === 'options') {
+                  return (
+                    <SurveyOptions
+                      key={element.id}
+                      element={element as ZetkinSurveyOptionsQuestionElement}
+                      onChange={(value) => {
+                        handleResponseChange(element.id, value);
+                      }}
+                    />
+                  );
+                }
+              } else {
+                return (
+                  <Box key={element.id} mb={1} mt={2}>
+                    <Typography variant="h6">
+                      {element.text_block.header}
+                    </Typography>
+                    <Typography>{element.text_block.content}</Typography>
+                  </Box>
```

Comment:
This can be handled by `SurveyTextBlock`, which will happen automatically if this entire `elements.map()` block is replaced with `<SurveyElements/>`.